pytorch_lightning/trainer/trainer.py

```diff
@@ -1700,10 +1700,6 @@ def world_size(self) -> int:
         # some training types define a world size
         return getattr(self.training_type_plugin, "world_size", 1)
 
-    @property
-    def should_rank_save_checkpoint(self) -> bool:
-        return self.training_type_plugin.should_rank_save_checkpoint
-
```

Comment:
since this is publicly exposed, this needs to go through a deprecation process

Reply:
noted in summary with context

Reply:
thanks for adding the context! i believe the initial removal was incorrect, and we should honor the deprecation process

Reply:
Resolving to merge as this PR is just addressing the merge mistake from #9901. Open an issue or comment on #9433 for this.